# Code and scripts for research papers  

 ðŸ“ƒ A repository of repositories containing code and scripts used in research articles, papers and preprints

View as a [web page](https://rngoodman.github.io/research-paper-code/) or [Github repository](https://github.com/rngoodman/research-paper-code).

## 2025

### [Shotgun metagenomic analysis of the oral microbiomes of children with noma reveals a novel disease-associated organism](https://doi.org/10.1101/2025.06.24.661267)
Michael Olaleye, Angus M O'Ferrall, **Richard N. Goodman**, Deogracia W Kabila, Miriam Peters, Gregoire Falq, Joseph Samuel, Donal Doyle, Diana Gomez, Gbemisola Oloruntuyi, Shafiu Isah, Adeniyi S Adetunji, Elise N. Farley, Nicholas J Evans, Mark Sherlock, Adam P. Roberts, Mohana Amirtharajah, Stuart Ainsworth **bioRxiv**  2025.06.24.661267; doi: https://doi.org/10.1101/2025.06.24.661267

> [Repository of code for paper](https://github.com/rngoodman/noma-metagenomics)
* [Within Noma dataset analysis - Table 1B, Figure 1A, Figure S1](https://rngoodman.github.io/noma-metagenomics/code/Noma_swab_vs_saliva.html)
* [Within healthy dataset analysis](https://rngoodman.github.io/noma-metagenomics/code/Healthy_vs_healthy.html)
* [Noma vs healthy dataset analysis - Table 1A, Figures 1B, 2 and 3, Figures S2-S4](https://rngoodman.github.io/noma-metagenomics/code/Noma_vs_healthy.html)

### [Genomic comparison of highly related pairs of E. coli and K. pneumoniae isolated from faeces and blood of the same neonatal patients hospitalized with fever in Dar es Salaam, Tanzania](https://doi.org/10.1101/2025.04.28.650962)
**Richard N. Goodman**, Sabrina J. Moyo, Ilinca Memelis, Aakash Khanijau, Joel Manyahi, Upendo O. Kibwana, Said Aboud, BjÃ¸rn Blomberg, Nina Langeland, Adam P. Roberts. **bioRxiv** 2025.04.28.650962; doi: https://doi.org/10.1101/2025.04.28.650962

> [Repository of code for paper](https://github.com/rngoodman/blood-faecal-genomic-comparison)
* [Figure 1](https://rngoodman.github.io/blood-faecal-genomic-comparison/vignettes/1_ANI_and_SNP_distance_matrices.html) - ANI and SNP distance matrices
* [Figure 2](https://rngoodman.github.io/blood-faecal-genomic-comparison/vignettes/2-core_genome_phylogenetic_tree.html) - core genome phylogenetic tree
* [Figure 3](https://rngoodman.github.io/blood-faecal-genomic-comparison/vignettes/3_variant_calling_heatmaps_with_breseq.html) - variant calling heatmaps with breseq
* [Figure 4](https://rngoodman.github.io/blood-faecal-genomic-comparison/vignettes/4_circular_whole_genome_comparisons.html) - circular whole genome comparisons
* [Figure 5](https://rngoodman.github.io/blood-faecal-genomic-comparison/vignettes/5_analysing_virulence_factor_determinants.html) - analysing virulence factor determinants

## 2024

### [Molecular mechanisms of re-emerging chloramphenicol susceptibility in extended-spectrum beta-lactamase-producing Enterobacterales](https://doi.org/10.1038/s41467-024-53391-2)
Graf, F.E., **Goodman, R.N.**, Gallichan, S. et al. **Nature Communications** 15, 9019 (2024). https://doi.org/10.1038/s41467-024-53391-2

> [Repository of code for paper](https://github.com/FEGraf/CHL-Malawi)
* [Figure 4](https://github.com/FEGraf/CHL-Malawi/tree/main/Fig4_and_SFig7_Co_occurence_analysis) - Co-occurence analysis of AMR genes
